### ISSUE TITLE
Do not attempt to transition before routing is ready

### ISSUE THREAD

**Issue by marcoow** [simplabs/ember-simple-auth#1539]
Currently, the `ApplicationRouteMixin` [maps the session service's events to methods on the mixin (and thus the application route) in `init`](https://github.com/simplabs/ember-simple-auth/blob/master/addon/mixins/application-route-mixin.js#L79). This leads to the problem that when the session is authenticated in an acceptance test before any route was visited (and the routing system is ready), the session service's event will be triggered which will cause a method on the application route to be called which will attempt a transition although the routing system is not ready yet. See [this writeup of the series of steps by @0xadada](https://github.com/simplabs/ember-simple-auth/pull/1519#issuecomment-359837396).

We could maybe only map the session service's events to methods on the application route (via the `ApplicationRouteMixin`) in the `activate` method (as opposed to `init`) which I assume would only be called after the routing system is ready but that might potentially lead to errors in edge cases… This needs some further investigation but is generally something that should be cleaned up.

**Comment by alexander-alvarez** [comment]
Spent the last couple of days researching this because it's affecting our application.

How can I help?

**Comment by 0xadada** [comment]
@alexander-alvarez  what does your test look like?

**Comment by alexander-alvarez** [comment]
@0xadada https://github.com/alexander-alvarez/ember-cli-qunit-service-registering/compare/572fae2#diff-6902ce6b3192309e4080d20541f6fbfd

**Comment by marcoow** [comment]
This should not happen when you're using the new acceptance test helpers introduced with 1.5.0. In order to fix this, the first thing to try would be moving the [mapping of session service events from the `init` method](https://github.com/simplabs/ember-simple-auth/blob/master/addon/mixins/application-route-mixin.js#L79) to the `activate` method I think - if anyone wants to jump in and try that.

**Comment by 0xadada** [comment]
@alexander-alvarez I'm confused. Your test component invalidates the session upon `init` hook, then tests that `yes` is output, when `no` _should_ be output because the session is not authenticated (because you called `invalidate`).

**Comment by 0xadada** [comment]
@alexander-alvarez per @marcoow, try the new test helper:
`tests/integration/components/my-component-test.js`:
```js
import { authenticateSession } from 'ember-simple-auth/test-support';
```
then
```js
-this.owner.register('service:session', Service.extend({
-      isAuthenticated: true,
-      invalidate() {
-        assert.ok(true);
-    }
-}));
+authenticateSession({
+        userId: 1,
+        otherData: 'some-data'
+});
```

**Comment by alexander-alvarez** [comment]
@0xadada  yes I understand. The example test case is just a POC to assert whether my service or the actual service was being called.  So if I get a session authenticated error, the service I've defined isn't being called, but if everything is all green, then my service is being called (which should happen).

sorry for the confusion brought up by the test semantics.

**Comment by 0xadada** [comment]
@alexander-alvarez 
> the service I've defined
as in, the service you stubbed in the test?

**Comment by alexander-alvarez** [comment]
correct. The "right-ness" I was evaluating was "here's this thing i've defined in the test, it should be used" not the specific logic of the test case.

**Comment by marcoow** [comment]
fixed by #1547
